src/base/utils/fs.cpp

```diff
@@ -49,17 +45,58 @@
 #include <unistd.h>
 #endif
 
+#include <QByteArray>
 #include <QDebug>
 #include <QDir>
 #include <QDirIterator>
 #include <QFile>
 #include <QFileInfo>
+#include <QIODevice>
 #include <QStorageInfo>
 #include <QRegularExpression>
 
 #include "base/bittorrent/torrenthandle.h"
 #include "base/global.h"
 
+Utils::Fs::FileOutputIterator::FileOutputIterator(QIODevice &device, const int bufferSize)
+    : m_device {&device}
+    , m_buffer {std::make_shared<QByteArray>()}
+    , m_bufferSize {bufferSize}
+{
+    m_buffer->reserve(bufferSize);
+}
+
+Utils::Fs::FileOutputIterator::~FileOutputIterator()
+{
+    m_device->write(*m_buffer);
+    m_buffer->clear();
+}
+
+Utils::Fs::FileOutputIterator &Utils::Fs::FileOutputIterator::operator=(const char c)
+{
+    m_buffer->append(c);
```

Comment:
Isn't QIODevice already buffered as well?

Reply:
I tried without any buffering (just `m_device->putChar(c);`) and it leads to ~3x worse run time, I guess it is overhead from function calls.